bukkit/src/main/java/net/william278/huskhomes/network/RedisMessenger.java

```diff
@@ -28,11 +27,13 @@ public void initialize(@NotNull HuskHomes implementor) {
         redisWorker = new RedisWorker(implementor.getSettings());
         redisWorker.initialize();
 
-        new Thread(() -> {
-            try (Jedis jedis = redisWorker.jedisPool.getResource()) {
-                jedis.subscribe(new JedisPubSub() {
+        CompletableFuture.runAsync(() -> {
```

Comment:
Let's run this in a `new Thread() ->` rather than through the future pool. In the future, I'll be largely moving away from using the future pool for various performance/stability reasons and running this on a named thread task is better for now.

Reply:
If I'm not wrong with the CompletableFuture#runAsync you can pass also an executor that is your custom thread pool. For the external library I can create a new PR in the future and add the library as a dependency.

Reply:
> If I'm not wrong with the CompletableFuture#runAsync you can pass also an executor that is your custom thread pool. 

There's a number of issues with this approach I've learnt through the dev process -- most notably that it can lead to thread exhaustion in certain cases and hanging tasks on server close. I'd like to switch to what I do with HuskTowns and have platform-implementing async scheduler support, and eliminate join calls from the code (see HuskTowns' TaskRunner.java class and the Bukkit implementation of this for details). That's the plan -- though not for this PR of course.

> For the external library I can create a new PR in the future and add the library as a dependency.

This sounds good.